sds-go/go/scanner_test.go

```diff
@@ -462,6 +482,43 @@ func TestPartialRedact(t *testing.T) {
 			str:     "here card ****39, this one should match, but this second one 382448 should not as it's not prefixed by card",
 			rules: []RuleMatch{{
 				RuleIdx:           0,
+				ReplacementType:   ReplacementTypePartialStart,
+				StartIndex:        10,
+				EndIndexExclusive: 16,
+				ShiftOffset:       0,
+			}},
+		},
+	}
+
+	runTest(t, scanner, testData)
+}
+
```

Comment:
Looks like we're missing a test for Hash redaction (and thus to test `ReplacementTypeHash`), could I ask you to add one, please?

Reply:
yes good catch

Reply:
the only problem is to not test the hash :)

Reply:
done